apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -107,56 +107,71 @@ defmodule OMG.Watcher.BlockGetter do
     {:noreply, state}
   end
 
-  @decorate measure_start()
-  def handle_cast(
-        {:apply_block,
-         %BlockApplication{
-           transactions: transactions,
-           number: blknum,
-           eth_height: eth_height
-         } = to_apply},
-        state
-      ) do
-    with {:ok, _} <- Core.chain_ok(state),
-         tx_exec_results = for(tx <- transactions, do: OMG.State.exec(tx, :ignore)),
-         {:ok, state} <- Core.validate_executions(tx_exec_results, to_apply, state) do
-      _ =
-        to_apply
+  def handle_continue({:execute_transactions, block_application}, state) do
+    tx_exec_results = for(tx <- block_application.transactions, do: OMG.State.exec(tx, :ignore))
+
+    case Core.validate_executions(tx_exec_results, block_application, state) do
+      {:ok, state} ->
+        block_application
         |> Core.ensure_block_imported_once(state)
         |> Enum.each(&DB.Transaction.update_with/1)
 
-      state = run_block_download_task(state)
+        {:noreply, state, {:continue, {:run_block_download_task, block_application}}}
 
-      {:ok, db_updates_from_state} = OMG.State.close_block(eth_height)
+      {{:error, _} = error, new_state} ->
+        :ok = update_status(new_state)
+        _ = Logger.error("Invalid block #{inspect(block_application.number)}, because of #{inspect(error)}")
+        {:noreply, new_state}
+    end
+  end
 
-      {state, synced_height, db_updates} = Core.apply_block(state, to_apply)
+  def handle_continue({:run_block_download_task, block_application}, state),
```

Comment:
I don't like it,  `run_block_download_task` does too `close_and_apply_block`
its more confusing.

if you want to cut it more, use private functions and cut it in a different way.

Reply:
I don't understand your first argument - it's a `sequence` of actions that occur explicitly.

Reply:
when i see `handle_continue({:run_block_download_task,` 
then I conclude that there will only be a download block and nothing else

Reply:
yep, that's exactly what happens in this scope, only :run_block_download_task.
and the return `:continue` implies _go to the next step_.

Reply:
that's what I mean, there's another step